pkg/controllers/dsa/controller.go

```diff
@@ -170,13 +159,12 @@ func (c *controller) UpdateDaemonSet(rawObj client.Object, ds *apps.DaemonSet) (
 		updated = true
 	}
 
-	if dp.Spec.NodeSelector == nil {
-		dp.Spec.NodeSelector = map[string]string{"kubernetes.io/arch": amd64}
-	} else {
-		dp.Spec.NodeSelector["kubernetes.io/arch"] = "amd64"
-	}
 	if !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, dp.Spec.NodeSelector) {
-		ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector
+		if len(dp.Spec.NodeSelector) > 0 {
+			ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector
+		} else {
+			ds.Spec.Template.Spec.NodeSelector = defaultNodeSelector
```

Comment:
nit: we shouldn't set ds node selector and updated=true if ds node selector is equal to defaultNodeSelector.

Reply:
Ah! OK.!